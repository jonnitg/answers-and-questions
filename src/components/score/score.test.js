import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Score from './score';

const resetPhases = jest.fn();

afterAll(cleanup);

it('renders score correctly', () => {
  const { getByText } = render(
    <Score questions={[]} answers={{}} resetPhases={resetPhases} />
  );
  const scoreTextElement = getByText(/questions/i);
  expect(scoreTextElement).toHaveTextContent('You got 0 right of 0 questions');
});

it('clicks button', () => {
  const { getByText } = render(
    <Score questions={[]} answers={{}} resetPhases={resetPhases} />
  );
  const scoreButtonElement = getByText(/reset test/i);
  fireEvent.click(scoreButtonElement);
  expect(resetPhases.mock.calls.length).toEqual(1);
});

it('matches snapshot', () => {
  const tree = renderer
    .create(<Score questions={[]} answers={{}} resetPhases={resetPhases} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
