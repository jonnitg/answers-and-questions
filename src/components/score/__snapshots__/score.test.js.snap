// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-AxjAm bGbvop"
>
  <div
    className="sc-AxirZ ghRTLL"
  >
    <div
      className="sc-AxiKw coLlqs"
    />
  </div>
</div>
`;
